tests/lib/package_utils.rb

```diff
@@ -50,15 +50,38 @@ def test_not_compatible_arch
     refute(PackageUtils.compatible?(pkg))
   end
 
-  def test_get_binary_url
+  def test_get_binary_url_old_hash
     pkg = Class.new(Package)
-    pkg.name = 'test_package'
+    pkg.name = 'hello_world_chromebrew'
     pkg.instance_eval do
-      version '1.0'
+      version '1.1'
       binary_compression 'tar.zst'
-      binary_sha256({ ARCH.to_sym => '0000000000000000000000000000000000000000000000000000000000000000' })
+      binary_sha256({
+        aarch64: '29185a6c4a8ecc3532606649f5831e37cd977c24bfd4fb1c12328ca5ba966ff1',
+         armv7l: '29185a6c4a8ecc3532606649f5831e37cd977c24bfd4fb1c12328ca5ba966ff1',
+           i686: '6e46c31245e9e17eeecf03d61454bb7312b6e769bfaf352ced527ee93cc62518',
+         x86_64: '83f674b3e8fe29c7e761ce6adc27fd7df927d985140ff6d1ae64a0046339322b'
+      })
+    end
+    package_file_id = { aarch64: '137956464', armv7l: '137956464', i686: '137956442', x86_64: '137956370' }
+    assert_equal("https://gitlab.com/chromebrew/binaries/-/package_files/#{package_file_id[ARCH.to_sym]}/download", PackageUtils.get_url(pkg))
+  end
+
+  def test_get_binary_url_new_hash
+    pkg = Class.new(Package)
+    pkg.name = 'hello_world_chromebrew'
+    pkg.instance_eval do
+      version '1.1'
+      binary_compression 'tar.zst'
+      binary_sha256({
+        aarch64: '827f9794864aa76f4c99fd31f989077f1fa65771386f73db30a7681842f8736d',
+         armv7l: '827f9794864aa76f4c99fd31f989077f1fa65771386f73db30a7681842f8736d',
+           i686: 'ff0942c505b04982fed187bcda123adead37b3ac2dcfd7e2f0543ca179e81df6',
+         x86_64: '3081f1f25950c91f75db41095b644a2783987a3a7ef2832fc2b85bf138bb006f'
+      })
     end
-    assert_equal("https://gitlab.com/api/v4/projects/26210301/packages/generic/test_package/1.0_#{ARCH}/test_package-1.0-chromeos-#{ARCH}.tar.zst", PackageUtils.get_url(pkg))
+    package_file_id = { aarch64: '137956527', armv7l: '137956527', i686: '137956521', x86_64: '137956520' }
+    assert_equal("https://gitlab.com/chromebrew/binaries/-/package_files/#{package_file_id[ARCH.to_sym]}/download", PackageUtils.get_url(pkg))
```

Comment:
Use a variable for `https://gitlab.com/chromebrew/binaries/-/package_files`?

Reply:
Same situation as before-- if the url changes to anything other than gitlab, this whole thing is useless, and I don't see us changing from the `chromebrew/binaries` repository to anything else anytime soon? Also, tests are supposed to be isolated from one another so sharing a variale would break that.

Reply:
This url is actually used more than twice.  A best practice in any development work is to follow the DRY (don't repeat yourself) principle.

Reply:
It's only used in this file twice-- I get what you're saying, but I think repeating something twice is acceptable.

Reply:
It's also used in `lib/package_utils.rb`.  This is even more reason to make it a constant.

Reply:
I strongly disagree. Adding an additional source-wide constant for a string that appears three times in two files is an unnecessary abstraction that would negatively impact code quality. Avoiding repetition is good, but this is an overzealous application of that principle.

Reply:
I wouldn't consider following best practices overzealous.  Think about the next time you might need to use the same url somewhere else and it might make a bit more sense.

Reply:
Fair enough. As it stands, I still believe an abstraction is unnecessary and would worsen code quality. However, if this changes in the future, i.e. we start referring to the gitlab repo url more frequently, I will add an abstraction if it is beneficial.

Reply:
Thank you, I appreciate the compromise. Let's see how this goes.